fe1-web/src/features/connect/screens/ConnectOpenScan.tsx

```diff
@@ -72,19 +83,28 @@ const ConnectOpenScan = ({ navigation }: IPropTypes) => {
     }
   };
 
-  return QrWasScanned ? (
-    <View style={containerStyles.centeredY} />
-  ) : (
+  return showScanner ? (
     <View style={containerStyles.centeredXY}>
       <QrReader delay={300} onError={handleError} onScan={handleScan} style={{ width: '30%' }} />
       <WideButtonView
         title={STRINGS.general_button_cancel}
         onPress={() => {
-          setQrWasScanned(true);
-          navigation.navigate(STRINGS.connect_unapproved_title);
+          setShowScanner(false);
+
+          // if we have an active lao, this was an additional connection and thus we navigate (back)
+          // to the organization user screen
+          if (laoId) {
+            navigation.navigate(STRINGS.app_navigation_tab_user, {
+              screen: STRINGS.organization_navigation_tab_user,
+            });
+          } else {
+            navigation.navigate(STRINGS.connect_unapproved_title);
+          }
         }}
```

Comment:
consider extracting function from the display template ?

Reply:
👍